src/pymor/models/neural_network.py

```diff
@@ -73,6 +76,81 @@ def _compute_solution(self, mu=None, **kwargs):
 
             return U
 
+    class NeuralNetworkInstationaryModel(Model):
+        """Class for models of instationary problems that use artificial neural networks.
+
+        This class implements a |Model| that uses a neural network for solving.
+
+        Parameters
+        ----------
+        T
+            The final time T.
+        Nt
+            The number of time steps.
+        neural_network
+            The neural network that approximates the mapping from parameter space
+            to solution space. Should be an instance of
+            :class:`~pymor.models.neural_network.FullyConnectedNN` with input size that
+            matches the (total) number of parameters and output size equal to the
+            dimension of the reduced space.
+        parameters
+            |Parameters| of the reduced order model (the same as used in the full-order
+            model).
+        output_functional
+            |Operator| mapping a given solution to the model output. In many applications,
+            this will be a |Functional|, i.e. an |Operator| mapping to scalars.
+            This is not required, however.
+        products
+            A dict of inner product |Operators| defined on the discrete space the
+            problem is posed on. For each product with key `'x'` a corresponding
+            attribute `x_product`, as well as a norm method `x_norm` is added to
+            the model.
+        estimator
+            An error estimator for the problem. This can be any object with
+            an `estimate(U, mu, m)` method. If `estimator` is
+            not `None`, an `estimate(U, mu)` method is added to the
+            model which will call `estimator.estimate(U, mu, self)`.
+        visualizer
+            A visualizer for the problem. This can be any object with
+            a `visualize(U, m, ...)` method. If `visualizer`
+            is not `None`, a `visualize(U, *args, **kwargs)` method is added
+            to the model which forwards its arguments to the
+            visualizer's `visualize` method.
+        name
+            Name of the model.
+        """
+
+        def __init__(self, T, Nt, neural_network, parameters={}, output_functional=None,
+                     products=None, estimator=None, visualizer=None, name=None):
+
+            super().__init__(products=products, estimator=estimator, visualizer=visualizer, name=name)
+
+            self.__auto_init(locals())
+            self.solution_space = NumpyVectorSpace(neural_network.output_dimension)
+            self.linear = output_functional is None or output_functional.linear
```

Comment:
I guess the `linear`-attribute does not really make sense for a neural network Model (also in the stationary case). What would that mean? In any case it should not only depend on the `output_functional`.